packages/common/src/ui/layout/tables/components/Header/Header.tsx

```diff
@@ -68,17 +70,49 @@ export const HeaderCell = <T extends DomainObject>({
       sortDirection={isSorted ? direction : false}
     >
       {sortable ? (
-        <TableSortLabel
-          hideSortIcon={false}
-          active={isSorted}
-          direction={direction}
-          IconComponent={SortDescIcon}
+        <Box
+          flexDirection="row"
+          display="flex"
+          justifyContent="space-between"
+          alignItems="center"
         >
-          <Header column={column} />
-        </TableSortLabel>
+          <TableSortLabel
+            hideSortIcon={false}
+            active={isSorted}
+            direction={direction}
+            IconComponent={SortDescIcon}
+          >
+            <Header column={column} />
+          </TableSortLabel>
+
+          {column?.onChangeWidth && (
+            <Box
+              sx={{
+                cursor: 'col-resize',
+                backgroundColor: 'transparent',
+              }}
+              draggable
+              onDragStart={e => {
+                setX(e.clientX);
+              }}
+              onDragEnd={e => {
+                column?.onChangeWidth &&
+                  column.onChangeWidth(
+                    Math.max(column.width + (e.clientX - x), column.minWidth)
+                  );
+
+                setX(e.clientX);
```

Comment:
Just went with a drag-then-resize rather then resizing in real time - easier.. though having it realtime is probably nicer

Reply:
the icon jumping back into place, and then the column resizing is a bit weird at first. ideally, on press a vertical line would show over the top of the table which gets dragged with the icon. I don't know if this is possible though

Reply:
Yeah it is.

- I think real-time column movements are just going to be too laggy. Though, maybe not and I'm overreacting - *especially* since we don't use editable columns in the larger list 🤔  .. basically my initial thought was that the types of cells we're going to have - different inputs, drop downs, dates etc will make it a little render heavy. But where those cells are used there aren't many rows 🤔 
- I went to google sheets and this 'style' of drag then resize is used - but you're right in that it uses an indicator.
- I *think* this should be significantly snappier on a built version so the lag between drag and resize should be a lot smaller
- It's probably worth giving the real time resize a proper goon a built version. Could also likely do it with refs and/or pure css though I suspect that might take a little longer
- I've done a quick attempt at the indicator using a react-table code sandbox as an example: https://codesandbox.io/s/empty-silence-rkukq?file=/src/App.js which seems to be OK. Only thing left to do would be getting the height of the actual table 🤔  -  using:
```
        &:active {
          &:before {
            content: "";
            position: absolute;
            background-color: red;
            z-index: 1000;
            width: 1px;
            min-height: 100vh;
          }
        }
```
- I've mainly been basing my assumptions of the column resize based on the react-table example - already with 20 rows and very basic data the table seems to be a bit janky in the code sandbox - though, a production version would likely feel fine. But increasing to 100 lines and it feels really bad and since this table seems to be relatively simple in term of what it's rendering I assumed real-time resize would be out of the question. But again, I think it's probably worth a shot?